### PR TITLE
feat: add scotch build scripts

### PR DESCRIPTION
**Pull request by j8xixo12**
Added scotch build script, this script compatible with `OSX` and `x86 Linux`.

### CODE REVIEW

File: scripts/build.d/scotch [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+OBJ		= .o
+
+MAKE		= make
+AR		= ar
+ARFLAGS		= -ruv
+CAT		= cat
+CCS		= gcc
+CCP		= mpicc
+CCD		= gcc
+CFLAGS		= -O3 -fPIC -DCOMMON_PTHREAD_BARRIER -DCOMMON_PTHREAD -DSCOTCH_CHECK_AUTO -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_TIMING_OLD -DSCOTCH_RENAME -DCOMMON_FILE_COMPRESS_BZ2 -DCOMMON_FILE_COMPRESS_GZ
```
Comment (tai271828) [inline scripts/build.d/scotch:31]: I am not sure if the indent is on purpose, isn't it?

File: scripts/build.d/scotch [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+CLIBFLAGS	=
+LDFLAGS		= -lz -lm -pthread -lbz2
```
Comment (tai271828) [inline scripts/build.d/scotch:33]: ditto

File: scripts/build.d/scotch [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+CP		= cp
+LEX		= flex -Pscotchyy -olex.yy.c
+LN		= ln
+MKDIR		= mkdir
+MV		= mv
+RANLIB		= ranlib
```
Comment (tai271828) [inline scripts/build.d/scotch:39]: ditto

**Review by tai271828 — COMMENTED** [review]
Thank you for the amazing work. I am looking forward to scotch for a while as well \o/ can't wait for it.

I raise some bike-shedding coding style comment. If they are expected, I am +1 on landing the code when CI passes.

File: scripts/build.d/scotch [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -e
+
+pkgname=scotch
+pkgbranch=${VERSION:-master}
+pkgfull=${pkgname}-${pkgbranch}
+
+# # unpack (clone / pull)
```
Comment (yungyuc) [inline scripts/build.d/scotch:9]: Why do you have two `#` in this line?

File: scripts/build.d/scotch [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+syncgit https://gitlab.inria.fr/scotch ${pkgname} ${pkgbranch}
```
Comment (yungyuc) [inline scripts/build.d/scotch:10]: For SCOTCH, I prefer to use released tarball, which is easier to upgrade.

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: scripts/build.d/scotch (hunk 2 above)
Comment (j8xixo12) [reply scripts/build.d/scotch:33]: ditto

File: scripts/build.d/scotch (hunk 3 above)
Comment (j8xixo12) [reply scripts/build.d/scotch:39]: ditto

File: scripts/build.d/scotch (hunk 4 above)
Comment (j8xixo12) [reply scripts/build.d/scotch:9]: typo, it will be fixed in next commit.

File: scripts/build.d/scotch (hunk 5 above)
Comment (tai271828) [reply scripts/build.d/scotch:10]: If tarball is also considered, `git submodule` may be a solution for fetching a batch of source by default.

For more details let me write the idea down with this discussion thread https://github.com/solvcon/devenv/discussions/30

Comment (yungyuc) [reply scripts/build.d/scotch:10]: Tarballs and submodules solve different things.  Tarballs may be used for offline building, and supports separate download and build.

I am not sure how well submodules play with explicit `git clone` and tarballs.  I'd guess it's either or?  If that's true, I would like to have tarballs and sacrifice submodules.

**Comment by yungyuc** [comment]
@j8xixo12 is it ready for a second review?

**Comment by j8xixo12** [comment]
@yungyuc Yes, it is ready for second review.

File: scripts/build.d/scotch (hunk 5 above)
Comment (j8xixo12) [reply scripts/build.d/scotch:10]: In my opinion, if we want to use tarball, devenv need another function to deal with it.
Maybe I can file an issue to record the requirement?

Comment (yungyuc) [reply scripts/build.d/scotch:10]: I see the created issue #32.  It looks good.

**Review by yungyuc — APPROVED** [review]

**Pull request merged**